var/spack/repos/builtin/packages/libsharp2/package.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Libsharp2(AutotoolsPackage):
+    """Libsharp2 is a code library for spherical harmonic transforms (SHTs) and
+    spin-weighted spherical harmonic transforms, which evolved from the libpsht
+    library. Because the upstream repository has no tags or releases, this
+    package tracks the versions published together with HEALPix releases."""
+
+    variant("openmp", default=True, description="Build with openmp support")
+    variant("mpi", default=True, description="Build with MPI support")
+    variant("pic", default=True, description="Generate position-independent code (PIC)")
+    variant("fast_math", default=True, description="Enable recommended fast-math optimisations")
+
+    homepage = "https://gitlab.mpcdf.mpg.de/mtr/libsharp"
+    git = "https://gitlab.mpcdf.mpg.de/mtr/libsharp.git"
+
+    version("3.82.0", sha256="47629f057a2daf06fca3305db1c6950edb9e61bbe2d7ed4d98ff05809da2a127")
+
+    conflicts("libsharp")
+
+    depends_on("autoconf", type="build")
+    depends_on("mpi", when="+mpi")
+
+    configure_directory = "src/common_libraries/libsharp"
+
+    def url_for_version(self, version):
+        major, minor, patch = version
+        return f"https://sourceforge.net/projects/healpix/files/Healpix_{major}.{minor}/healpix_cxx-{major}.{minor}.{patch}.tar.gz/download"
+
```

Comment:
not an expert, but isn't this the data about healpix? also above version is 3.82.0 which is healpix and the sha does not indeed correspond to the one of healpix which is set above to `9c1b0b...`

Reply:
Though the error message is correct: if I download the Healpix package at version v3.82 fro Sourceforge as per instructions of their documentation and then check the sha I get this

```
[michele@d-h-122 Downloads]$ shasum -a 256 Healpix_3.82_2022Jul28.tar.gz
47629f057a2daf06fca3305db1c6950edb9e61bbe2d7ed4d98ff05809da2a127  Healpix_3.82_2022Jul28.tar.gz
```

so I think this is the correct sha to use here